### ISSUE TITLE
PG 14 upgrade

### ISSUE THREAD

**Issue by pyramation** [constructive-io/pgsql-parser#92]
[libpg-query@14.0.0](https://www.npmjs.com/package/libpg-query/v/14.0.0)

**Comment by eljefedelrodeodeljefe** [comment]
@pyramation do you have plans on integrating it? If you need help what would be a one-liner statement of what needs to be done?

**Comment by pyramation** [comment]
When I attempted this, the string type was changed — so some of the messages and node parsing needed to be updated. Would love to take a PR for this — and it would also allow for 15 to happen as well.

**Comment by nith2001** [comment]
@pyramation Hello! Not sure what you were referring to by the string type, but does it related to the `libpg-query-node` repository needing to upgrade to V15? Relevant PR [here](https://github.com/pyramation/libpg-query-node/pull/27).

**Comment by pyramation** [comment]
> @pyramation Hello! Not sure what you were referring to by the string type, but does it related to the `libpg-query-node` repository needing to upgrade to V15? Relevant PR [here](https://github.com/pyramation/libpg-query-node/pull/27).

yep, the output from `libpq-query-node` produces a different JSON structure than before for some of the primitive node types — which I believe was the raw string node. This one is particularly everywhere in the `pgsql-parser` deparser code. One idea for a first attempt is to preparse the output JSON so it conforms to the current codebase and current deparser, and later we can update the deparser.

**Comment by nith2001** [comment]
Ahh, I see. So the idea would be to write some code in `libpq-query-node` to take the output from the original `libpq-query` code in Version 15, pre-parse it, and return that to `pgsql-parser` whenever it is called?

If so, that would definitely work, but would complications arise in other codebases where the `libpq-query-node` library is being used? Or does that not matter because the only way that would happen is for other codebases to update their version?

**Edit**: Also ngl, I've almost never worked in C++ so I feel like my learning curve would be a tad steep to try and help contribute to `libpg-query-node`. If there's anything I can do to help on the Javascript side of things on `pgsql-parser` like writing tests, I'd be down to help.

**Comment by pyramation** [comment]
PG17 is ready! https://www.npmjs.com/package/pgsql-parser

I'll try to get older ones working too, but figured this would be worth sharing here

windows, linux, mac, all supported! No N-API or node-gyp, all WASM!

cc @eljefedelrodeodeljefe
